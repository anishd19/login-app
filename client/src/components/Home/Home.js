import React, { Component } from 'react';
import { Navbar } from '../Navbar';
import './Home.scss';

class Home extends Component {
  render() {
    return (
      <div className="container is-two-quarters">
        <Navbar />
        <div className="card">
          <div className="card-content">
            <p className="title">
              Welcome to Home Page
            </p>
            <p className="subtitle">
              Coded by Anish Dhamodaran
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;