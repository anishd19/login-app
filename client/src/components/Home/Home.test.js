import React from "react";
import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Home from "./Home";
import store from "../../store";

afterEach(cleanup);

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders html elements", () => {
  const { getByText } = render(
    <Provider store={store}>
      <Home />
    </Provider>
  );

  expect(getByText(/Welcome to Home Page/i)).toBeTruthy();
  expect(getByText(/Coded by Anish Dhamodaran/i)).toBeTruthy();
});
