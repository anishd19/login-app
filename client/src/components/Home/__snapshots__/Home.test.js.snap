// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container is-two-quarters"
>
  <div
    className="navbar"
  >
    <div
      className="container"
    >
      <div
        className="navbar-brand"
      >
        <a
          className="navbar-item"
          href="#"
        >
          <div
            className="logo"
          >
            Login App
          </div>
        </a>
      </div>
      <div
        className="navbar-menu"
      >
        <div
          className="navbar-end"
        >
          <a
            className="navbar-item"
            onClick={[Function]}
          >
            Log out
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <p
        className="title"
      >
        Welcome to Home Page
      </p>
      <p
        className="subtitle"
      >
        Coded by Anish Dhamodaran
      </p>
    </div>
  </div>
</div>
`;
